homeassistant/components/hunterdouglas_powerview/coordinator.py

```diff
@@ -27,18 +31,39 @@ def __init__(
     ) -> None:
         """Initialize DataUpdateCoordinator to gather data for specific SmartPlug."""
         self.shades = shades
+        self.maintenance = False
         super().__init__(
             hass,
             _LOGGER,
             name=f"powerview hub {hub_address}",
-            update_interval=timedelta(seconds=60),
+            update_interval=UPDATE_INTERVAL_DEFAULT,
         )
 
     async def _async_update_data(self) -> PowerviewShadeData:
         """Fetch data from shade endpoint."""
+
+        if self.maintenance is True:
+            self.update_interval = UPDATE_INTERVAL_DEFAULT
+            self.maintenance = False
+            _LOGGER.debug("Polling returned to %s", UPDATE_INTERVAL_DEFAULT)
+
         async with async_timeout.timeout(10):
             shade_entries = await self.shades.get_resources()
-        if not shade_entries:
+
+        if isinstance(shade_entries, bool):
+            # hub returns boolean on a 204/423 empty response (maintenance)
+            # continual polling results in inevitable error
+            # restart of hub takes between 3-5 minutes and generally between 12am-3am
+            _LOGGER.debug(
```

Comment:
What happens if you try to control a shade during the maintenance window?

Reply:
not sure - given it is something that seems to happen at random times between 12 and 3 in the morning not something that i thought of testing

what behaviour are you expecting and how we would handle it?

I expect the command would fail with a timeout error.

Maybe we add a check if self.maintenance is True when we send a command and write a warning to the log ?

Reply:
I'm actually thinking we should still raise `UpdateFailed` with a message that the hub is doing maint here since it will put all the shades in an unavailable state which accurately reflects what is going on here.

Reply:
We could also leave the polling interval the same so it recovers as soon as maint is finished, which would eliminate some of the complexity here.

Reply:
> I'm actually thinking we should still raise `UpdateFailed` with a message

What I would expect to see in the logs in that case would be something like
12:01 - Successful polling
12:02: Update Failed - Hub undergoing maint
12:03: Update Failed - Hub undergoing maint
12:04: Update Failed - Hub timed out
12:05 - Successful polling

Was trying to minimise the errors in the log's but you're right they would be unavailable for that small period
I just prefer to not have errors in the log if you can avoid / handle them hence the debug and not an update failed

What about keeping it as debug (or even change to a warning so the user sees it) and keep polling at default?

Reply:
Only the first update failure will be logged. I think it's ok to log that since it's actually failing for a small window. I don't think we should hide that. It's a unfortunate design problem with the hub, and I don't want to dissuade users from complaining about it the vendor by hiding it in the logs.